src/main/java/com/exadel/discount/service/impl/JwtGenerationServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.exadel.discount.service.impl;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.exadel.discount.service.JwtGenerationService;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+@Service
+@Setter
+@Slf4j
+public class JwtGenerationServiceImpl implements JwtGenerationService {
+    private final String ROLES_CLAIM_NAME = "role";
+
+    @Value("${jwt.encryption.key}")
+    private String TOKEN_ENCRYPTION_KEY;
+    @Value("${jwt.refresh.role}")
+    private String REFRESH_ROLE;
+    @Value("${jwt.refresh.expiration}")
+    private long REFRESH_TOKEN_EXPIRATION_TIME;
+    @Value("${jwt.access.expiration}")
+    private long ACCESS_TOKEN_EXPIRATION_TIME;
+
+    @Override
+    public String generateAccessToken(UserDetails userDetails) {
+        log.debug("preparing an access token creation data");
+        return buildToken(
+                userDetails.getUsername(),
+                ACCESS_TOKEN_EXPIRATION_TIME,
+                userDetails.getAuthorities()
+                        .stream()
+                        .map(GrantedAuthority::getAuthority)
+                        .collect(Collectors.joining()));
```

Comment:
Why are you joining roles? if it will be multiple it will produce invalid concatenated role

Reply:
It was resolved in commit "Commit with amendments №2".